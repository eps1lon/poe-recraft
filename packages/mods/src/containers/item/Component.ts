export default interface Component<T, B> {
  parent: T;
  builder(): B;
  any(): boolean;
}
