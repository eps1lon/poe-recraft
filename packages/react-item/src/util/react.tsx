import * as React from 'react';

// text nodes dont need keys
export const CommaSeparator = () => ', ';

export const LineBreak = (key: string) => <br key={key} />;

export function intersperse(
  nodes: React.ReactNode[],
  renderSeparator: (key: string) => React.ReactNode = CommaSeparator,
) {
  if (nodes.length === 0) {
    return [];
  }

  return nodes.slice(1).reduce(
    (joined: React.ReactNode[], element, index) => {
      // push instead of concat's returnval because concat is expensive
      joined.push(renderSeparator(`separator-${index}`), element);
      return joined;
    },
    [nodes[0]],
  );
}

export interface Props {
  renderSeparator: (key: string) => React.ReactNode;
}

export class Intersperse extends React.PureComponent<Props> {
  public render() {
    const { children, renderSeparator } = this.props;
    if (Array.isArray(children) && children.length > 1) {
      return children.reduce((prev, cur, i) => {
        if (prev && cur) {
          return [prev, renderSeparator(`separator-${i}`), cur];
        } else if (cur) {
          return cur;
        } else {
          return prev;
        }
      });
    } else {
      return children;
    }
  }
}
