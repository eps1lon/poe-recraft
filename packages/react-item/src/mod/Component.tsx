import { PureComponent } from 'react';

import { Props as StatProps } from '../stat/Stat';

export interface Props {
  id: string;
  primary: number;
  stats: StatProps[];
}

export default class Mod extends PureComponent<Props> {
  public render() {
    return 'Mod';
  }
}
