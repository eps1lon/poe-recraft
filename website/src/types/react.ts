import { ComponentType } from 'react';

export type PropsType<C> = C extends ComponentType<infer P> ? P : never;

export type PartialProps<C, K extends keyof PropsType<C>> = Pick<
  PropsType<C>,
  K
>;
