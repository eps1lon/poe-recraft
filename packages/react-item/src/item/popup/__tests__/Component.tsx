import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Component from '../ItemPopup';

it('should display Mind Brow Hubris Circlet', () => {
  expect(
    renderer
      .create(
        <Component
          item={{
            base: {
              name: 'Hubris Circlet',
            },
            name: 'Mind Brow',
            rarity: 'rare',
            energy_shield: {
              value: 200,
              augmented: true,
            },
            quality: 6,
            shaper: true,
            requirements: {
              level: {
                value: 69,
              },
              intelligence: {
                value: 154,
              },
            },
            explicitStats: [
              'Place an additional Mine',
              'Socketed Gems are Supported by Level 18 Remote Mine',
              '16% increased Effect of Chill',
              'Socketed Gems are Supported by Level 18 Hypothermia',
              '+1 to Level of Socketed Minion Gems',
              '+67 to maximum Life',
              '18% increased Stun and Block Recovery',
              'Adds 5 to 89 Lightning Damage to Spells',
            ],
          }}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

it('should display Noble Axe', () => {
  expect(
    renderer
      .create(
        <Component
          item={{
            base: {
              name: 'Noble Axe',
            },
            rarity: 'normal',
            requirements: {
              level: {
                value: 65,
              },
              strength: {
                value: 113,
              },
            },
            aps: {
              value: 130,
            },
          }}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
