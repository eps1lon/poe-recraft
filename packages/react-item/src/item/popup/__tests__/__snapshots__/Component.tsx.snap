// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display Mind Brow Hubris Circlet 1`] = `
<div
  className="poe-item rare"
  style={
    Object {
      "maxWidth": 742.39350912779,
      "minWidth": 262.88032454361,
      "width": "auto",
    }
  }
>
  <header
    className="double-line shaper"
  >
    <span
      className="name-left"
    />
    <div
      className="item-name"
    >
      <span
        className="lc"
      >
        Mind Brow
      </span>
    </div>
    <div
      className="item-name typeLine"
    >
      <span
        className="lc"
      />
    </div>
    <span
      className="name-right"
    />
  </header>
  <section
    className="body"
  >
    <div
      className="displayProperty"
    >
      <span
        className="lc"
      >
        <span>
          Quality
        </span>
        : 
        <span
          className="augmented"
        >
          6%
        </span>
      </span>
    </div>
    <div
      className="displayProperty"
    >
      <span
        className="lc"
      >
        <span>
          energy shield
        </span>
        : 
        <span
          className="augmented"
        >
          200
        </span>
      </span>
    </div>
    <div
      className="separator"
    />
    <div
      className="requirements"
    >
      <span
        className="lc"
      >
        <span>
          Requires
        </span>
         
        <span>
          <span>
            Level
          </span>
        </span>
         
        <span>
          <span
            className="simple"
          >
            69
          </span>
        </span>
        , 
        <span>
          <span
            className="simple"
          >
            154
          </span>
        </span>
         
        <span>
          <span>
            Int
          </span>
        </span>
      </span>
    </div>
    <div
      className="separator"
    />
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        Place an additional Mine
      </span>
    </div>
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        Socketed Gems are Supported by Level 18 Remote Mine
      </span>
    </div>
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        16% increased Effect of Chill
      </span>
    </div>
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        Socketed Gems are Supported by Level 18 Hypothermia
      </span>
    </div>
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        +1 to Level of Socketed Minion Gems
      </span>
    </div>
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        +67 to maximum Life
      </span>
    </div>
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        18% increased Stun and Block Recovery
      </span>
    </div>
    <div
      className="explicitMod"
    >
      <span
        className="lc"
      >
        Adds 5 to 89 Lightning Damage to Spells
      </span>
    </div>
  </section>
</div>
`;

exports[`should display Noble Axe 1`] = `
<div
  className="poe-item normal"
  style={
    Object {
      "maxWidth": 742.39350912779,
      "minWidth": 262.88032454361,
      "width": "auto",
    }
  }
>
  <header
    className=""
  >
    <span
      className="name-left"
    />
    <div
      className="item-name typeLine"
    >
      <span
        className="lc"
      />
    </div>
    <span
      className="name-right"
    />
  </header>
  <section
    className="body"
  >
    <div
      className="displayProperty"
    >
      <span
        className="lc"
      >
        <span>
          Attacks per Second
        </span>
        : 
        <span
          className="simple"
        >
          1.30
        </span>
      </span>
    </div>
    <div
      className="separator"
    />
    <div
      className="requirements"
    >
      <span
        className="lc"
      >
        <span>
          Requires
        </span>
         
        <span>
          <span>
            Level
          </span>
        </span>
         
        <span>
          <span
            className="simple"
          >
            65
          </span>
        </span>
        , 
        <span>
          <span
            className="simple"
          >
            113
          </span>
        </span>
         
        <span>
          <span>
            Str
          </span>
        </span>
      </span>
    </div>
  </section>
</div>
`;
